packages/devtools_app/lib/src/table.dart

```diff
@@ -957,6 +1014,36 @@ class _TableRowState<T> extends State<TableRow<T>>
         widget.onExpansionCompleted();
       }
     });
+
+    if (widget.searchMatchesNotifier != null) {
+      searchMatches = widget.searchMatchesNotifier.value;
+      addAutoDisposeListener(widget.searchMatchesNotifier, () {
+        final isPreviousMatch = searchMatches.contains(widget.node);
+        final isNewMatch =
+            widget.searchMatchesNotifier.value.contains(widget.node);
+        searchMatches = widget.searchMatchesNotifier.value;
+
+        // We only want to rebuild the row if it the match status has changed.
+        if (isPreviousMatch ^ isNewMatch) {
+          setState(() {});
```

Comment:
why not use `!=` instead of `^`?

also can we store whether there is a match in the state object to avoid the empty setState and make it clear what the state we are tracking is?